src/main/java/duke/storage/FileSystem.java

```diff
@@ -0,0 +1,117 @@
+package duke.storage;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.exception.InvalidDateTimeException;
+import duke.helper.TaskList;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * FileSystem class that handles the saving and loading of tasks
+ */
+public class FileSystem {
+    private File file;
+
+    /**
+     * Constructor for FileSystem class
+     *
+     * @param relFilePath files's location
+     * @throws IOException If file cannot be created
+     */
+    public FileSystem(String relFilePath) throws IOException {
+        String dirPath = relFilePath.split("/")[0];
+        File dir = new File(dirPath);
+        file = new File(relFilePath);
+
+        if (!dir.exists()) {
+            dir.mkdirs();
+        }
+
+        if (!file.exists()) {
+            file.createNewFile();
+        }
+    }
+
+    /**
+     * Updates the file according to the tasks stored on the list
+     *
+     * @param taskList the TaskList that contains all the tasks that will be stored in the file
+     */
+    public void updateFile(TaskList taskList) {
+        try {
+            file.delete();
+            file.createNewFile();
+            ArrayList<Task> tasks = taskList.getTasks();
+
+            for (Task t : tasks) {
+                appendToFile(t.toString());
+            }
+        } catch (IOException e) {
+            e.toString();
+        }
+    }
+
+    /**
+     * Loads the tasks from the file into the TaskList
+     *
+     * @return an ArrayList that contains all the tasks from the file
+     * @throws FileNotFoundException If the file cannot be found
+     */
+    public ArrayList<Task> loadFromFile() throws FileNotFoundException, InvalidDateTimeException {
+        Scanner scanner = new Scanner(file);
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        while (scanner.hasNextLine()) {
+            String task = scanner.nextLine();
```

Comment:
this logic that is repeated in a loop may be abstracted into its own function/class so that in the event of an error, it is easy to find and fix.